File: Noiembrie-2023/Week_no4/Meditație 29.11.2023/Teorema_fundamentala_a_aritmeticii.java
import java.util.*;
public class Teorema_fundamentala_a_aritmeticii{
	public static void main(String[] args){
		Scanner scan = new Scanner( System.in );
		int n = scan.nextInt();
		System.out.println("Teorema Fundamentala a Aritmeticii:\n");
		int i = 2;
		while (!(n==1)){
			int fm = 0;
			while(n%i==0){
				fm +=1;
				n/=i;
			}
			if( fm!=0 )
				System.out.println(i + "^"+ fm);
			i++;
		}
	}
}

File: Noiembrie-2023/Week_no4/Meditație 29.11.2023/Teorema_fundamentala_a_aritmeticii.py
def main():

	print("Teorema Fundamentala a Aritmeticii:\n")
	n = int(input("N = "))
	i = 2
	while not n==1:
		fm=0
		while n%i==0:
			fm += 1
			n //=i
		if fm: 
			print("%d^%d\n" %(i, fm))
		i += 1
main()